src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useContext, useState } from 'react';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  Alert, DataTable, Icon, IconButton,
+} from '@edx/paragon';
+import { Info, Visibility } from '@edx/paragon/icons';
+import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
+import { BulkEmailContext } from '../../bulk-email-context';
+import { getScheduledBulkEmailThunk } from './data/thunks';
+import messages from './messages';
+import ViewEmailModal from '../ViewEmailModal';
+
+function flattenScheduledEmailsArray(emails) {
+  return emails.map((email) => ({
+    id: email.id,
+    task: email.task,
+    taskDue: new Date(email.taskDue).toLocaleString(),
+    ...email.courseEmail,
+    targets: email.courseEmail.targets.join(', '),
+  }));
+}
+
+function BulkEmailScheduledEmailsTable({ intl }) {
+  const { courseId } = useParams();
+  const [{ scheduledEmailsTable }, dispatch] = useContext(BulkEmailContext);
+  const [viewModal, setViewModal] = useState({
+    isOpen: false,
+    messageContent: {},
+  });
+
+  const fetchTableData = useCallback((args) => {
+    dispatch(getScheduledBulkEmailThunk(courseId, args.pageIndex + 1));
+  }, []);
+
+  const handleViewEmail = (row) => {
+    setViewModal({
+      isOpen: true,
+      messageContent: {
+        subject: row.original.subject,
+        requester: row.original.sender,
+        created: '',
+        email: {
+          html_message: row.original.htmlMessage,
+        },
+        sent_to: row.original.targets,
+      },
+    });
+  };
+  if (scheduledEmailsTable.errorRetrievingData) {
+    return (
+      <Alert variant="danger" icon={Info}>
+        <Alert.Heading>{intl.formatMessage(messages.bulkEmailScheduledEmailsTableErrorHeader)}</Alert.Heading>
+        <p>{intl.formatMessage(messages.bulkEmailScheduledEmailsTableError)}</p>
+      </Alert>
+    );
+  }
+  return (
+    <>
+      {viewModal.isOpen && (
+        <ViewEmailModal
+          isOpen={viewModal.isOpen}
+          setModalOpen={(open) => setViewModal({ isOpen: open })}
+          messageContent={viewModal.messageContent}
+        />
+      )}
+      <DataTable
```

Comment:
Does the data table have some way to re-send the API call to refresh the contents? If not, should we use the same mechanism the pending task table uses today to refresh the data in the table at a regular interval?

Reply:
I can create one: https://paragon-openedx.netlify.app/components/datatable/#bulk-actions

I would prefer to allow the user to manually refresh the table. I can also have the submit email button refresh the table on a successfully scheduled email.

I can add it to this PR or add it to the coming work around this table.

Reply:
Ive added this to the AC of the next subtask.

Reply:
Thanks, TJ!